mithril-signer/src/database/repository/signed_beacon_repository.rs

```diff
@@ -0,0 +1,312 @@
+            .await
+            .unwrap();
+
+        assert_eq!(to_filter, available_entities);
+    }
+
+    #[tokio::test]
+    async fn filter_out_nothing_if_previously_signed_entities_doesnt_match_passed_entities() {
+        let connection = Arc::new(main_db_connection().unwrap());
+        let repository = SignedBeaconRepository::new(connection.clone());
+
+        let time_point = TimePoint::dummy();
+        insert_signed_beacons(
+            &connection,
+            SignedBeaconRecord::fakes(&[(
+                Epoch(1941),
+                vec![SignedEntityType::MithrilStakeDistribution(
+                    time_point.epoch - 2,
+                )],
+            )]),
+        );
+        let to_filter = all_signed_entity_type_for(&time_point);
+
+        let available_entities = repository
+            .filter_out_already_signed_entities(to_filter.clone())
+            .await
+            .unwrap();
+        assert_eq!(to_filter, available_entities);
+    }
+
+    #[tokio::test]
+    async fn filter_out_everything_if_previously_signed_entities_match_all_passed_entities() {
+        let connection = Arc::new(main_db_connection().unwrap());
+        let repository = SignedBeaconRepository::new(connection.clone());
+
+        let to_filter = all_signed_entity_type_for(&TimePoint::dummy());
+        insert_signed_beacons(
+            &connection,
+            to_filter
+                .iter()
+                .map(|entity| SignedBeaconRecord::fake(Epoch(4872), entity.clone()))
+                .collect(),
+        );
+
+        let available_entities = repository
+            .filter_out_already_signed_entities(to_filter.clone())
+            .await
+            .unwrap();
+        assert_eq!(Vec::<SignedEntityType>::new(), available_entities);
+    }
+
+    #[tokio::test]
+    async fn filter_out_partially_if_some_previously_signed_entities_match_passed_entities() {
+        let connection = Arc::new(main_db_connection().unwrap());
+        let repository = SignedBeaconRepository::new(connection.clone());
+
+        let time_point = TimePoint::dummy();
+        let signed_beacons = [
+            SignedEntityType::MithrilStakeDistribution(time_point.epoch),
+            SignedEntityType::CardanoTransactions(
+                time_point.epoch,
+                time_point.chain_point.block_number,
+            ),
+        ];
+        insert_signed_beacons(
+            &connection,
+            signed_beacons
+                .iter()
+                .map(|entity| SignedBeaconRecord::fake(Epoch(4872), entity.clone()))
+                .collect(),
+        );
+
+        let to_filter = all_signed_entity_type_for(&time_point);
+        let available_entities = repository
+            .filter_out_already_signed_entities(to_filter.clone())
+            .await
+            .unwrap();
+
+        let expected: Vec<SignedEntityType> = to_filter
+            .iter()
+            .filter(|entity| !signed_beacons.contains(entity))
```

Comment:
This code is the same as the one in the code of filter_out_already_signed_entities.
Is this test is really relevant if we use the same code in both sides ?

Reply:
You're right, i changed the test to use hard coded lists.